### PR TITLE
Refactor daemon simulation test

### PR DESCRIPTION
**Pull request by emlowe**
The test `test_daemon_simulation` would bring up an entire full stack including a timelord and then only tests that you can send a specific message to the node thru the daemon. This test doesn't seem particularly useful and has been consolidated with the original "full_system" test that was `test_simulation_1`

It was noticed that the test `test_daemon_simulation` and the test `test_simulation_1` are essentially the same test. They both bring up the entire full stack of multiple nodes, farmer, harvest, timelord, vdf_clients, and the daemon. There was only one minor difference in that the `test_simulation_1` didn't connect any nodes to the daemon. Since these tests are heavy and very slow to bring up everything, they have been consolidated into a single test `test_full_system` that brings up the entire system, runs the original tests, then connects a node to the daemon and runs a few additional daemon specific tests.

The `timelord_launcher.py` code needed some refactor in order to allow the `setup_vdf_clients` code to work properly multiple times in the same test run. Previously this would have issues as the multiple runs would stomp on a global variable in `timelord_launcher.py`

This now allows multiple "simulation" tests to be run in the same test run, and we can remove the check in the `simulation` fixture.

A new simple daemon test was added that brings up the daemon and a node - it makes sure the node connects to the daemon and then runs a singular test that the daemon can be used as a communication hub and send messages to/from the node.

Todo:
- [x] new daemon test should verify the payload of the return message and not just that "something" was returned.

### CODE REVIEW

**Comment by emlowe** [comment]
Coverage complaints are due to missing existing tests around the signal handling for vdf client teardown. Can be added later.

**Review by arvidn — DISMISSED** [review]
lgtm. maybe at least one other person should review too, before landing

File: tests/conftest.py
Comment (almogdepaz) [inline tests/conftest.py:952]: i think we can delete this

**Review by almogdepaz — DISMISSED** [review]

File: tests/conftest.py
Comment (altendky) [reply tests/conftest.py:952]: I'm going to go ahead and make this a raised concern since we at least shouldn't leave it sitting commented out.

**Review by altendky — DISMISSED** [review]

File: tests/conftest.py
Comment (emlowe) [reply tests/conftest.py]: Removed

**Comment by emlowe** [comment]
Close and reopen for CI

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:32]: Make this a method?  Seems pretty closely related.

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:91]: It feels like this and other stuff should be wrapped up by the class.  Basically, instead of providing a locking mechanism and hoping folks use it, apply the locking mechanism yourself with helpful abstractions that can be called instead.

Or maybe this whole function is a method maybe?

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:28]: And associated call-site simplifications.

```suggestion
    lock: asyncio.Lock = field(default_factory=asyncio.Lock)
    stopped: bool = False
```

**Review by altendky — CHANGES_REQUESTED** [review]
Mostly it seems like the new abstraction `VDFClientProcessMgr` could pull in some more methods and code.

File: chia/timelord/timelord_launcher.py
Comment (emlowe) [reply chia/timelord/timelord_launcher.py]: Added

File: chia/simulator/setup_services.py
Comment (altendky) [inline chia/simulator/setup_services.py:439]:
```suggestion
    process_mgr = VDFClientProcessMgr()
```

File: chia/simulator/setup_services.py
Comment (altendky) [inline chia/simulator/setup_services.py:470]:
```suggestion
    process_mgr = VDFClientProcessMgr()
```

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:123]:
```suggestion
    process_mgr = VDFClientProcessMgr()
```

**Review by altendky — CHANGES_REQUESTED** [review]

**Comment by emlowe** [comment]
Set to draft while debugging test problem

**Comment by emlowe** [comment]
Finally got this passing 100/100 in a repeated test run.

File: chia/simulator/setup_nodes.py
Comment (altendky) [inline chia/simulator/setup_nodes.py:395]: Does `shared_b_tools.config["self_hostname"]` have the wrong value?

File: chia/simulator/setup_nodes.py
Comment (altendky) [inline chia/simulator/setup_nodes.py:446]: Pass by name?

File: chia/simulator/setup_services.py
Comment (altendky) [inline chia/simulator/setup_services.py:473]:
```suggestion
    process_mgr = VDFClientProcessMgr()
```

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:129]: Given that we're changing runtime timelord stuff, let's create an ephemeral testnet for this when we get close to the end.  We can leave this conversation open until we take care of that.

File: tests/simulation/config.py
Comment (altendky) [inline tests/simulation/config.py:5]: Diff noise?

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/simulator/setup_nodes.py
Comment (emlowe) [reply chia/simulator/setup_nodes.py:395]: No I just used a variable at the top `self_hostname = shared_b_tools.config["self_hostname"]`

File: chia/simulator/setup_services.py
Comment (emlowe) [reply chia/simulator/setup_services.py:466]: Yes, I originally just did `await` and while this works on macos fine, it did not on linux. It resulted in a hang as the await never returned - however, it's possible it would work now that I properly await the process ending after the kill. So I then tried a `cancel` and `await` and this ended up raising CancelledError in the await call - but I know it's Cancelled as I cancelled it so this just seemed less code than another try handler.
Note, I also just tried `cancel` but sometimes the process ends before the cancel is processed resulting in teardown errors about pending tasks. Since this is all teardown code, it seems consuming any exceptions while waiting for the task to finish after cancelling seemed reasonable to me

File: chia/simulator/setup_nodes.py
Comment (emlowe) [reply chia/simulator/setup_nodes.py]: Changed

File: chia/timelord/timelord_launcher.py
Comment (emlowe) [reply chia/timelord/timelord_launcher.py]: Removed

File: chia/timelord/timelord_launcher.py
Comment (emlowe) [reply chia/timelord/timelord_launcher.py]: Fixed to inside lock (nice catch)

File: tests/simulation/config.py
Comment (emlowe) [reply tests/simulation/config.py]: yup, fixed

File: chia/simulator/setup_nodes.py
Comment (emlowe) [reply chia/simulator/setup_nodes.py:345]: Digging into this seems outside the scope and I really didn't want get caught up in a big mess here

File: chia/simulator/setup_services.py
Comment (altendky) [reply chia/simulator/setup_services.py:466]: 1) I think most exceptions should be reported, we don't want silent failures we don't know about on shutdown.
2) The cancellation might not be yours...  but we won't deal with that in asyncio, at least not for now.
3) (actually actionable) I don't think that `asyncio.gather()` is a good way to express 'consume exceptions from a single call'.  I'd definitely rather see a `try`/`except`.  And I don't think using `asyncio.gather()` should have fixed anything vs. a direct `await`.

File: chia/timelord/timelord_launcher.py
Comment (altendky) [inline chia/timelord/timelord_launcher.py:61]:
```suggestion
    raise FileNotFoundError("Cannot find vdf_client binary. Is Timelord installed?  See install-timelord.sh")
```

Comment (emlowe) [reply chia/timelord/timelord_launcher.py]: fixed

File: chia/timelord/timelord_launcher.py
Comment (emlowe) [reply chia/timelord/timelord_launcher.py:103]: This loops around and just starts up new processes as the old ones exit (for any reason) - so I'm not sure it helps much to add that in?

**Review by altendky — APPROVED** [review]

File: chia/simulator/setup_nodes.py
Comment (arvidn) [reply chia/simulator/setup_nodes.py:141]: what's even more suspicious is that support for v1 was dropped. It's possible that this parameter doesn't do anything anymore. There is a fixture still, that's always 2 now.

**Review by arvidn — APPROVED** [review]

**Pull request merged**
